### DISCUSSION TITLE
Traces not showing up in Errors tab after going from log4j to log4j2

### DISCUSSION THREAD

**Discussion by kingbeau** [glowroot/glowroot#947] [Q&A]
First, awesome tool!  We have recently upgraded all the things in our application (not glowroot) and after doing so, no longer get error traces showing up in the Errors tab.  There is one type of error showing up in the tab that is not logged by us explicitly and happens somewhere in Spring.  We are using 0.3.16 and the central collector.  Transactions show up correctly and the Queries tab is also populated.  I have seen the stack in the Thread Profile as well.  Double checked that "Error with throwable" was checked for the Logger plugin and also checked "Error without throwable" for fun.

The ownership of glowroot, the application, and the log files has changed but they all match.  Glowroot is able to connect to the collector and the log files in both the glowroot and glowroot-central directories are being populated.

The application upgraded to use log4j2 instead of log4j and still uses slf4j as a wrapper.  The logging levels and files are identical albeit in a different format for the different log4j versions.  I tried doing a test where `logger.error(message, throwable)` was hit and it shows up as an error in the log file but does not show up in the Errors tab of glowroot.  It does show up in the Transactions tab.  These are the logging dependencies:
```        
        <dependency>
            <groupId>org.slf4j</groupId>
            <artifactId>slf4j-api</artifactId>
            <version>1.7.36</version>
        </dependency>
        <dependency>
            <groupId>org.apache.logging.log4j</groupId>
            <artifactId>log4j-core</artifactId>
            <version>2.17.2</version>
        </dependency>
        <dependency>
            <groupId>org.apache.logging.log4j</groupId>
            <artifactId>log4j-slf4j-impl</artifactId>
            <version>2.17.2</version>
            <exclusions>
                <exclusion>
                    <artifactId>slf4j-api</artifactId>
                    <groupId>org.slf4j</groupId>
                </exclusion>
            </exclusions>
        </dependency>
```

Here is an example of one of the configured loggers in log4j2.xml
```
<?xml version="1.0" encoding="UTF-8" ?>
<Configuration status="WARN">
    <Appenders>
        <RollingFile
                name="error"
                fileName="/var/log/error.log"
                filePattern="/var/log/error.log.%i"
                append="true">
            <PatternLayout>
                <Pattern>%d{yyyy-MM-dd HH:mm:ss} %-5p %c{1}:%L - %m%n</Pattern>
            </PatternLayout>
            <Policies>
                <SizeBasedTriggeringPolicy size="10MB" />
            </Policies>
            <DefaultRolloverStrategy max="5" />
        </RollingFile>
    </Appenders>
    <Loggers>
        <Logger name="errorLogger" additivity="false" level="info">
            <AppenderRef ref="error"/>
        </Logger>
    </Loggers>
</Configuration
```

I've triple checked permissions and am reaching the end of my debugging ability.  Is there any way to turn on some more verbose logging in glowroot or is there a way to debug why this isn't making it into the Errors tab?

Any help is appreciated.  Thank you.

**Update**
I debugged the logging context a little and saw that only my initial server logging statements were making it in there.  No other message with or without Throwable we making it in.  I downgraded the log4j2 dependencies to 2.11.1 and the Error messages work as expected.  Played with trying to find the newest version of log4j2 that would work and only made it to 2.12.0.  2.12.1 works the same as 2.17.2 for me.  Did not see anything obvious in the log4j2 release notes that I thought would have an impact, but I also don't know what I was looking for.

**Update 2**
From the 2.12.1 log4j release notes I found this commit https://github.com/apache/logging-log4j2/commit/24adaff that looks like it lines up with what I'm seeing.  It changed to use the `log()` method of the Logger class instead of the `logMessage()` method that matches the `@Pointcut` definition on the Log4j2xAspect class.

**Comment by rsgillit** [comment]
I am facing same issue with Springboot, Tomcat, log4j2 2.17 onwards. JDK8/17 both.
I have upgraded glowroot jar to latest but still same.
Everything works just fine with log4j2 2.11.1
Is there going to be fix for this ?
